server/services/websockets.js

```diff
@@ -206,19 +214,232 @@ export default class Websockets {
+            .emit('collections.remove_user', {
+              event: event.name,
+              userId: event.userId,
+              collectionId: event.collectionId,
+            });
+
+          // tell any user clients to disconnect from the websocket channel for the collection
+          socketio.to(`user-${event.userId}`).emit('leave', {
+            event: event.name,
+            collectionId: event.collectionId,
+          });
+        }
+        return;
+      }
+      case 'collections.add_group': {
+        const group = await Group.findByPk(event.data.groupId);
+
+        // the users being added are not yet in the websocket channel for the collection
+        // so they need to be notified separately
+        for (const groupMembership of group.groupMemberships) {
+          socketio
+            .to(`user-${groupMembership.userId}`)
+            .emit('collections.add_user', {
+              event: event.name,
+              userId: groupMembership.userId,
+              collectionId: event.collectionId,
+            });
+
+          // tell any user clients to connect to the websocket channel for the collection
+          socketio.to(`user-${groupMembership.userId}`).emit('join', {
+            event: event.name,
+            collectionId: event.collectionId,
+          });
+        }
+        return;
+      }
+      case 'collections.remove_group': {
+        const group = await Group.findByPk(event.data.groupId);
+        const membershipUserIds = Collection.membershipUserIds(
+          event.collectionId
+        );
+
+        for (const groupMembership of group.groupMemberships) {
+          if (membershipUserIds.includes(groupMembership.userId)) {
+            // the user still has access through some means...
+            // treat this like an add, so that the client re-syncs policies
+            socketio
+              .to(`user-${groupMembership.userId}`)
+              .emit('collections.add_user', {
+                event: event.name,
+                userId: groupMembership.userId,
+                collectionId: event.collectionId,
+              });
+          } else {
+            // let users in the channel know they were removed
+            socketio
+              .to(`user-${groupMembership.userId}`)
+              .emit('collection.remove_user', {
+                event: event.name,
+                userId: groupMembership.userId,
+                collectionId: event.collectionId,
+              });
+
+            // tell any user clients to disconnect to the websocket channel for the collection
+            socketio.to(`user-${groupMembership.userId}`).emit('leave', {
+              event: event.name,
+              collectionId: event.collectionId,
+            });
+          }
+        }
+        return;
+      }
+      case 'groups.add_user': {
+        // do an add user for every collection that the group is a part of
+        const collectionGroupMemberships = CollectionGroup.find({
```

Comment:
Also hitting an error here, I think we want `findAll` ?

![image](https://user-images.githubusercontent.com/380914/76187512-c5ceb800-6192-11ea-89d6-5173f7e12ddb.png)